### PR TITLE
Add node 16 and remove 10 support

### PR DESCRIPTION
**Pull request by vmatyusGitHub**
## PR summary

Relates to: #254 

## PR Checklist

Please make sure that your PR fulfills the following requirements:

- [ ] The commit message follows the
[Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type  

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] New tests
- [x] Build/CI related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

## What is the current behavior?

Currently Node 10 is supported and 15 is also tested, 16 is not included.

## What is the new behavior?

Introduce Node 16 and remove 10 and 15 from supported versions.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

`>=12` would allow to have 15 as node version, so better if it specified per versions.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -32,7 +32,7 @@
   },
   "license": "Apache-2.0",
   "engines": {
-    "node": ">=10.0.0"
+    "node": ">=12.0.0 <15.0.0 || >=16.0.0"
```
Comment (eiri) [inline package.json:35]: this also allows to run on node v13.x which we do not support. why this change from recommended on the parent ticket `"node": "^12 || ^14 || ^16"` ?

**Review by eiri — CHANGES_REQUESTED** [review]
So I looked into this and the problem here is not our engine definition, but the fact that `ibm-cloud-sdk-core` sets node version too wide as `node >= 10`. Take a notice of "combined" word in the warning below.

```
ibm-cloud-sdk-core: Narrower "node" engine requirement needed: >=10
Combined "node" engine requirement needs to be narrower: ^12.0.0
```

To explain what's going on here: condition `^12 || ^14 || ^16` excludes range `13.x`, so it conflicts with core's condition "_anything_ that larger than 10.x". With your change the whole range from 12 to 16 covered and that complies with condition "larger than 10", so no warning, but this is a wrong fix, we want 13 excluded as it's unstable release.

So change "node" back to `^12 || ^14 || ^16` and change "check-packages" in "scripts" to `installed-check -e -d -v -i ibm-cloud-sdk-core` to ignore core in engine intersection check.

**Comment by vmatyusGitHub** [comment]
Considering there are 3 dependencies in the package and the remaining 2 has no engine, I do not think this check adds any value.
I also checked the operation and the whole check is weird.
It uses [intersect of versions](https://www.npmjs.com/package/@voxpelli/semver-set), but only returns the "lowest common dominator". And it accepts only when the intersect returns the same node version value in the `package.json`: https://github.com/voxpelli/node-installed-check-core/blob/4242ed5cffabbcfb3ab50ba6df786bf06e4a6197/lib/check-engine-versions.js#L64

So `Combined "node" engine requirement needs to be narrower: ^12.0.0` means it accepts only `"node": "^12.0.0"`

**Comment by eiri** [comment]
I agree that it's pretty useless tool and I don't think it's very good in what it's trying to do, consider, for example, that `^12 < 13.0.0` is not passing condition, but `^12 < 13.0.1`, so I'm fine with removing it and its this step from travis completely too.

**Comment by vmatyusGitHub** [comment]
I checked the tool, besides the engine version checks it also makes a package version assertion. So the locally used package version should comply with the predefined semantic version.
So, I think I will just skip the core currently and I will open an issue against the tool, and they might correct the engine check condition.

**Comment by vmatyusGitHub** [comment]
Reported the problem against the checker tool: https://github.com/voxpelli/node-installed-check-core/issues/18

**Review by eiri — APPROVED** [review]
legit

**Review by emlaver — APPROVED** [review]

**Comment by vmatyusGitHub** [comment]
Thank you for reviewing!

**Comment by vmatyusGitHub** [comment]
Now the package correction is ready: https://github.com/voxpelli/node-installed-check-core/issues/18
So this change can be reverted.

**Comment by ricellis** [comment]
Thanks for the pointer @vmatyusGitHub - we dropped `check-packages` in #280 but I'll take a look and see if it would add any value now.

**Pull request merged**